### ISSUE TITLE
QR Scan will not work with some logos inside of a Frame QR 

### ISSUE THREAD

**Issue by An-anonymous-coder** [GrapheneOS/Camera#470]
I noticed that GrapheneOS handles stylized QR codes very well, but some logos (such as the Target logo) break the QR code reader. Here is a demonstration of the same QR code with and without a logo:
![with_logo](https://github.com/user-attachments/assets/46daf551-b9a7-47f7-9d33-11ffb94c7f54)
![without_logo](https://github.com/user-attachments/assets/6585d638-b84e-4dd7-8348-87a011fa2aea)

The second one scans just fine, whereas the first one will refuse to scan.

This bug is present in version 74. 

P.S. This _is_ fixable, as the iOS camera is able to scan both. Good luck!

**Comment by MHShetty** [comment]
The library we are using to support various barcode formats isn't accepting issues for certain type of barcode formats not getting scanned unfortunately. Switching to another library could be done, but it could lead to more issues such as missing formats from what we currently offer and other edge cases not being covered that zxing doesn't cover

**Comment by MHShetty** [comment]
I'm currently aware of zxing (the one we're currently using) and ZBar. Will need to look into this to figure out if it's worth migrating to another barcode scan library. We're currently working on migrating different parts of our camera app to Jetpack Compose, so we might only be able to look into this once that gets completed.

Will notify you via this thread for further updates, thanks for your valuable time and help!

**Comment by MHShetty** [comment]
> Just found [this](https://github.com/markusfisch/BinaryEye) which scanned the QR code listed here, which is weird since it's the same library (ZXing)

It seems to be using a CPP port of the zxing library internally

**Comment by MHShetty** [comment]
> Could the GrapheneOS camera do the same?

Yes, but it would need testing and debugging in case of issues that arise to ensure nothing else breaks in the process and re-implementing a lot of things, just to support a specific edge case (i.e. scanning qr codes with circle target, whereas fancy QRs are mostly still supported by the app). 

Compared to the efforts that would be needed v/s what we would be providing to the users, resolving this issue might not be the top priority at the moment, but will be looked into once other issue/ongoing work with higher priority get addressed.

We'll surely notify you whenever we get back to working on this issue or have something to discuss or share. 

Thanks a lot for your valuable time and feedback!
